src/drivers/rc_epgsql.erl

```diff
@@ -0,0 +1,62 @@
+-module (rc_epgsql).
+
+-behaviour (gen_server).
+
+-export ([start_link/1]).
+-export ([init/1, handle_info/2, handle_cast/2, handle_call/3, terminate/2]).
+
+start_link(#{name := Name} = Args) ->
+  gen_server:start_link({local, Name}, ?MODULE, Args, []);
+
+start_link(Args) ->
+  gen_server:start_link({local, epgsql}, ?MODULE, Args, []).
+
+init(Args) ->
+  self() ! connect,
+  process_flag(trap_exit, true),
+  ValidArgs =
+    #{host         => maps:get(host, Args, "localhost"),
+      username     => maps:get(username, Args, os:getenv("USER")),
+      password     => maps:get(password, Args, ""),
+      database     => maps:get(database, Args, "test_db"),
+      reconnection => maps:get(reconnection, Args, {uniform, 4000})},
+  {ok, maps:merge(#{retries => 0, state => no_connection}, ValidArgs)}.
+
+handle_info(connect, #{reconnection := {uniform, Time}} = State) ->
+  #{host     := Host,
+    username := Username,
+    password := Pass,
+    database := Database} = State,
+  try
+    {ok, Pid} = epgsql:connect(Host, Username, Pass, [{database, Database}, {timeout, Time}]),
+    {noreply, State#{connection_ref => Pid, state => connected}}
+  catch
+    error:{badmatch, Reason} ->
```

Comment:
it would be better to have a case in epgsql:connect to handle errors, instead of catching a badmatch